src/hera.cpp

```diff
@@ -44,38 +44,57 @@ using namespace HeraVM;
 
 namespace {
 
-vector<uint8_t> sentinel(struct evm_context* context, vector<uint8_t> const& input)
-{
-#if HERA_DEBUGGING
-  cerr << "Metering (input " << input.size() << " bytes)..." << endl;
-#endif
-
-#if HERA_METERING_CONTRACT
-  evm_message metering_message = {
-    .address = { .bytes = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0xa } }, // precompile address 0x00...0a
+vector<uint8_t> callSystemContract(
+  struct evm_context* context,
+  const struct evm_address& address,
+  int64_t & gas,
+  vector<uint8_t> const& input
+) {
+  evm_message message = {
+    .address = address,
     .sender = {},
     .value = {},
     .input = input.data(),
     .input_size = input.size(),
     .code_hash = {},
-    .gas = -1, // do not charge for metering yet (give unlimited gas)
+    .gas = gas,
     .depth = 0,
     .kind = EVM_CALL,
     .flags = EVM_STATIC
   };
 
-  evm_result metering_result;
-  context->fn_table->call(&metering_result, context, &metering_message);
+  evm_result result;
+  context->fn_table->call(&result, context, &message);
 
   vector<uint8_t> ret;
-  if (metering_result.status_code == EVM_SUCCESS && metering_result.output_data)
-    ret.assign(metering_result.output_data, metering_result.output_data + metering_result.output_size);
+  if (result.status_code == EVM_SUCCESS && result.output_data)
+    ret.assign(result.output_data, result.output_data + result.output_size);
+
+  gas = result.gas_left - gas;
```

Comment:
Not `gas = result.gas_left`?

Reply:
Well I guess that can move to the caller. In `sentinel` I want to display the actual amount used (and later also charge for that).